core/src/test/java/dk/alexandra/fresco/lib/real/MathTests.java

```diff
@@ -342,4 +339,166 @@ public void test() throws Exception {
       };
     }
   }
+
+  public static class TestReciprocal<ResourcePoolT extends ResourcePool>
+      extends TestThreadFactory<ResourcePoolT, ProtocolBuilderNumeric> {
+
+    @Override
+    public TestThread<ResourcePoolT, ProtocolBuilderNumeric> next() {
+      List<BigDecimal> openInputs = Stream.of(.01, 0.1, 1.1, 2.1, 3.1, 4.1, 5.1, 10.1)
```

Comment:
Maybe create a shared vault of test data for these (and other) real tests - they seem to be chosen somewhat biased?

Reply:
Maybe. But the functions has different restrictions on input (size, sign, etc.), so I'm not sure how to do this in a nice way. For now, I have changed the test data so it's obvious that it was not generated randomly, but rather chosen to represent input of different order of magnitudes.